blog/_posts/2018-04-30-in-a-nutshell.md

```diff
@@ -182,6 +182,13 @@ that so far required advanced macro code to define. In particular:
 
  - There will be a way to do typeclass derivation a la [Kittens](https://github.com/milessabin/kittens), [Magnolia](https://github.com/propensive/magnolia), or [scalaz-deriving](https://gitlab.com/fommil/scalaz-deriving) that does not need macros. We are currently evaluating the alternatives. The primary goal is to develop a scheme that is easy to use and that performs well at both compile- and run-time. A second goal is generality, as long as it does not conflict with the primary goal.
 
+## Meta Programming in the Large
+
+The future Scala 3 macro design is intended to replace the existing def macros and the `scala.reflect` infrastructure. But there is another meta programming system that is quite complementary to it: [Scalameta](http://scalameta.org/) provides high-quality syntactic analysis and code generation tools which are separate from the Scala compiler. As the name implies, Scalameta is run at the meta level, that is, it takes programs as input and produces syntactic or semantic information or rewritten programs as output. A macro system, by contrast, is integrated in the language and expands programs as they are compiled. There are potential synergies between the two projects. To name but two possibilities:
```

Comment:
> syntactic **and semantic** analysis

With SemanticDB it's possible to attribute Scalameta syntax trees with symbols and soon also types.